src/discussions/posts/data/slices.js

```diff
@@ -127,10 +111,11 @@ const threadsSlice = createSlice({
     },
     deleteThreadSuccess: (state, { payload }) => {
       const { threadId } = payload;
-      const { topicId } = state.threads[threadId];
+      const { topicId } = state.threadsById[threadId];
       state.postStatus = RequestStatus.SUCCESSFUL;
-      state.topicThreadMap[topicId] = state.topicThreadMap[topicId].filter(item => item !== threadId);
-      delete state.threads[threadId];
+      state.threadsInTopic[topicId] = state.threadsInTopic[topicId].filter(item => item !== threadId);
+      state.pages = state.pages.map(page => page?.filter(item => item !== threadId));
```

Comment:
this will also not work as pages are being altered to object while fetching.

Reply:
I'm not sure I understand why you think pages is being altered to an object? It is still an array, and the tests dealing with this code are still passing.